test/SILGen/lazy_property_with_observers.swift

```diff
@@ -0,0 +1,58 @@
+// RUN: %target-swift-emit-silgen %s | %FileCheck %s
+
+class Foo {
```

Comment:
It's worthwhile testing with a struct as well, since the getter being mutating is a special case.